src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,78 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	public delegate void GamepadCallback (Gamepad gamepad);
+	public delegate void RawGamepadCallback (RawGamepad raw_gamepad);
+
+	private static GamepadMonitor instance;
+
+	private GenericSet<Gamepad?> gamepads;
+	private GenericSet<RawGamepad?> unmapped_raw_gamepads;
```

Comment:
This way you implemented this looks good, but we don't use it at all and as we are not writing a library it's better to have no unused code. Though, don't hesitate to keep it in a different branch to add it later when we will actually need it! :)